src/Hal.cpp

```diff
@@ -0,0 +1,17 @@
+#include <stdint.h>
+#include "Hal.h"
+
+Hal::Hal(const uint8_t input, const uint8_t output, const uint8_t low, const uint8_t high, const uint8_t rising, const uint8_t falling)
+    : GpioModeInput(input),
+      GpioModeOutput(output),
+      GpioLevelLow(low),
+      GpioLevelHigh(high),
+      GpioInterruptRising(rising),
+      GpioInterruptFalling(falling) {}
+
+void Hal::init(){};
+void Hal::term(){};
+void Hal::tone(uint8_t pin, unsigned int frequency, unsigned long duration){};
+void Hal::noTone(uint8_t pin){};
+void Hal::yield(){};
+uint8_t Hal::pinToInterrupt(uint8_t pin) { return pin; };
```

Comment:
What is the reason these methods are not pure virtual like the rest?

Reply:
They are not pure because on some platform they doesn't exist, and will simply be defined as nop. We can simply make them optional and implement them as not for platform that doesn't support them.
init and term are optional since they are `Hal`-specific and can or not be useful.

Reply:
OK, thanks for the clarification, this can remain as-is.